### PR TITLE
Include dev fixes

### PR DESCRIPTION
**Pull request by andrie**

### CODE REVIEW

File: R/assertions.R [hunk 1]
```diff
@@ -12,18 +12,14 @@ on_failure(is_input_id) <- function(call, env) {
 
 
 is_header <- function(x) {
-  is.null(x) || (is.character(x) && length(x) == 1 && !is.na(x))
+  is.na(x) || (is.character(x) && length(x) == 1)
 }
 
 on_failure(is_header) <- function(call, env) {
```
Comment (schloerke) [inline R/assertions.R:18]: IDK how this worked before

```suggestion
on_failure <- function(call, env) {
```

File: R/bucket_list.R [hunk 2]
```diff
@@ -64,7 +68,7 @@ is_add_rank_list <- function(x) {
 #'   shiny::runApp(app)
 #' }
 bucket_list <- function(
-  header = NULL,
+  header = NA,
```
Comment (schloerke) [inline R/bucket_list.R:71]: I (and my other packages) have a strong preference towards using `NULL`. 

I would prefer to keep as `NULL` but will leave it to you.

**Review by schloerke — CHANGES_REQUESTED** [review]
Why the change away from `NULL` and towards `NA`?

File: R/bucket_list.R (hunk 2 above)
Comment (andrie) [reply R/bucket_list.R]: I've reverted to the original behavour (`header = NULL`) and fixed the assertion to also deal with NA values.

File: R/assertions.R [hunk 3]
```diff
@@ -12,18 +12,14 @@ on_failure(is_input_id) <- function(call, env) {
 
 
 is_header <- function(x) {
-  is.null(x) || (is.character(x) && length(x) == 1 && !is.na(x))
+  is.null(x) || length(x) == 1 && (is.na(x) || is.character(x))
```
Comment (schloerke) [inline R/assertions.R:15]: Self reassurance that the order is kept as we expect it to be

```suggestion
  is.null(x) || (length(x) == 1 && (is.na(x) || is.character(x)))
```

File: R/bucket_list.R [hunk 4]
```diff
@@ -73,7 +77,8 @@ bucket_list <- function(
   orientation = c("horizontal", "vertical")
 ) {
 
-  # assert_that(is_header(header))
+  assert_that(is_header(header))
+  if (is.na(header)) header <- NULL
```
Comment (schloerke) [inline R/bucket_list.R:81]: Protect against `is.na(NULL) # logical(0)`

```suggestion
  if (isTRUE(is.na(header))) header <- NULL
```

**Review by schloerke — APPROVED** [review]
Minor tweaks, then LGTM!

**Pull request merged**